packages/node/src/indexer/indexer.manager.ts

```diff
@@ -134,6 +144,22 @@ export class IndexerManager {
     this.vm.on('console.log', (data) => getLogger('sandbox').info(data));
   }
 
+  private getStartBlockFromDataSources() {
+    const startBlocksList = this.project.dataSources
+      .filter(
+        (ds) =>
+          !ds.filter?.specName ||
+          ds.filter.specName === this.api.runtimeVersion.specName.toString(),
+      )
+      .map((item) => item.startBlock ?? 1);
+    if (startBlocksList.length === 0) {
+      logger.error(`Failed to get start block from valid data sources`);
```

Comment:
error message not very clear, try `Failed to find a valid datasource, Please check your endpoint if specName filter is used.`